src/slider/slider.component.ts

```diff
@@ -276,6 +282,18 @@ export class Slider implements AfterViewInit, ControlValueAccessor {
 	get disabled() {
 		return this._disabled;
 	}
+	/** Set to `true` for a readonly state. */
+	@Input() set readonly(v: boolean) {
+		this._readonly = v;
+		// for some reason `this.input` never exists here, so we have to query for it here too
+		const inputs = this.getInputs();
+		if (inputs && inputs.length > 0) {
+			inputs.forEach(input => input.readOnly = v);
+		}
+	}
+	get readonly() {
+		return this._readonly;
+	}
```

Comment:
> // for some reason `this.input` never exists here, so we have to query for it here too

That's expected, input probably hasn't been rendered when it was called. The entire slider component needs a rewrite imo. :/ Most likely will be done in next major version.